### PR TITLE
Replace `user.find` with `edit.find`

### LINKED ISSUE #1493: There no longer needs to be a distinction between edit.find and user.find
Now we have a separate user.find tag we can collapse these definitions.

### PR DESCRIPTION
**Pull request by AndreasArvidsson**
We should definitely not implement user namespace actions we're Talon already have defined them on internal namespaces.

Fixes #1493

### CODE REVIEW

File: apps/eclipse/eclipse_win.py [hunk 1]
```diff
@@ -42,9 +42,16 @@ def toggle_comment():
 
 @ctx.action_class("edit")
 class EditActions:
-    def find(text: str):
+    def find(text: str = None):
         actions.key("ctrl-f")
-        actions.insert(text)
+        if text:
+            actions.insert(text)
```
Comment (auscompgeek) [inline apps/eclipse/eclipse_win.py:48]: I believe this matches the default Windows implementation, right? Should we just remove the matching ones?

Comment (AndreasArvidsson) [reply apps/eclipse/eclipse_win.py:48]: Yes we should probably have an operating system defaults not every single application needs to specify this. I'm on it.

Comment (AndreasArvidsson) [reply apps/eclipse/eclipse_win.py]: Done

File: tags/find/find.py [hunk 2]
```diff
@@ -1,16 +1,27 @@
...
 
+ctx = Context()
 
-@mod.action_class
-class Actions:
-    def find(text: str):
-        """Finds text in current editor"""
+ctx_mac = Context()
+ctx_mac.matches = r"""
+os: windows
```
Comment (auscompgeek) [inline tags/find/find.py:10]: ?

File: tags/find/find.py [hunk 3]
```diff
@@ -1,16 +1,27 @@
...
+class EditActions:
+    def find(text: str = None):
+        actions.key("ctrl-f")
+        if text:
+            actions.insert(text)
+
+
+@ctx_mac.action_class("edit")
+class MacEditActions:
+    def find(text: str = None):
```
Comment (auscompgeek) [inline tags/find/find.py:24]: Assuming this is meant to target macOS, the only correct macOS generic implementation exists in `edit_mac.py`.

File: tags/find/find.py (hunk 2 above)
Comment (AndreasArvidsson) [reply tags/find/find.py]: wops copy paste error :D

File: tags/find/find.py (hunk 3 above)
Comment (AndreasArvidsson) [reply tags/find/find.py:24]: Okay I think we move these implementations to edit instead

**Comment by AndreasArvidsson** [comment]
@auscompgeek We now have a platform specific implementation in their corresponding edit files. There only exists application specific overrides where they differ.

**Review by nriley — APPROVED** [review]

File: apps/vscode/vscode.py [hunk 4]
```diff
@@ -169,6 +167,11 @@ class MacUserActions:
     def command_palette():
         actions.key("cmd-shift-p")
 
+    def find(text: str = None):
```
Comment (nriley) [inline apps/vscode/vscode.py:170]: This is defining an override for `user.find`, not `edit.find`.

**Review by nriley — COMMENTED** [review]
Sorry, missed this!

File: apps/vscode/vscode.py (hunk 4 above)
Comment (AndreasArvidsson) [reply apps/vscode/vscode.py:170]: Good catch!
I made a follow up here
https://github.com/talonhub/community/pull/1511

**Pull request merged**